server/src/test/java/org/opensearch/indices/replication/SegmentReplicationTargetTests.java

```diff
@@ -361,6 +392,111 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * This tests ensures that new files generated on primary (due to delete operation) are not considered missing on replica
+     * @throws IOException
+     */
+    public void test_MissingFiles_NotCausingFailure() throws IOException {
+        int docCount = 1 + random().nextInt(10);
+        List<Store.MetadataSnapshot> storeMetadataSnapshots = generateStoreMetadataSnapshot(docCount);
```

Comment:
Nitpick - please add comments here so that it's clear what the subsequent `get(0)` and `get(1)` fetch

Reply:
Done